python/revng/cli/_commands/daemon_self_test/__init__.py

```diff
@@ -0,0 +1,93 @@
+#
+# This file is distributed under the MIT License. See LICENSE.md for details.
+#
+
+import asyncio
+import os
+import sys
+from argparse import ArgumentParser
+from tempfile import TemporaryDirectory
+
+from revng.cli.commands_registry import Command, CommandsRegistry, Options
+from revng.cli.support import get_driver_options
+
+from .daemon_handler import DaemonProto, ExternalDaemonProto, InteractiveDaemonProto
+from .daemon_handler import NonInteractiveDaemonProto
+from .test import run_self_test
+
+
+class DaemonSelfTestCommand(Command):
+    FILTER_ENV = [
+        "STARLETTE_DEBUG",
+        "REVNG_NOTIFY_FIFOS",
+        "REVNG_ORIGINS",
+        "REVNG_DATA_DIR",
+        "REVNG_PROJECT_ID",
+    ]
+
+    def __init__(self):
+        super().__init__(
+            ("daemon-self-test",),
+            "Check if revng-daemon can produce all artifacts given an executable",
+        )
+
+    def register_arguments(self, parser: ArgumentParser):
+        parser.add_argument(
+            "-e",
+            "--external",
+            help="Use the specified external address instead of using the local daemon",
+        )
+        parser.add_argument(
+            "--revng-c",
+            action="store_true",
+            help="Enable additional checks, which assume revng-c is installed",
+        )
+        parser.add_argument(
+            "executable", metavar="EXECUTABLE", help="Executable to run the test with"
+        )
+
+    def run(self, options: Options):
+        args = options.parsed_args
+
+        if args.external is not None:
+            url = args.external
+            self.proto: DaemonProto = ExternalDaemonProto(url)
+        else:
+            temp_folder = TemporaryDirectory()
+            url = f"unix:{temp_folder.name}/daemon.sock"
+
+            interactive = args.gdb or args.lldb
+            driver_options = get_driver_options()
```

Comment:
Then `run` needs to be reworked, currently it will block until the subprocess has finished. In this case the daemon needs to be started without blocking.

Reply:
Reworked `run`